### PR TITLE
Fix: use vcard.url instead of foaf.openid for new contacts

### PR DESCRIPTION
**Pull request by VirginiaBalseiro**
- with new contacts, create a 'blank' node under `vcard.url`
- read from `foaf.openid` and `vcard.url`
- update tests

- [x] I've added a unit test to test for potential regressions of this bug.
- [ ] The changelog has been updated, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: components/contactsList/index.jsx [hunk 1]
```diff
@@ -124,10 +124,12 @@ function ContactsList() {
...
-  const contacts = profiles.map((p) => ({
-    thing: p,
-    dataset: addressBook,
-  }));
+  const contacts = profiles.map((p) => {
+    return {
+      thing: p,
+      dataset: addressBook,
+    };
+  });
```
Comment (ajacksified) [inline components/contactsList/index.jsx:132]: Could be cleaned up with an implicit return: 

```
const contacts = profiles.map((thing) => ({ thing, dataset: addressBook });
```

**Review by ajacksified — COMMENTED** [review]
Looks good, just a minor comment.

**Review by ajacksified — APPROVED** [review]

File: __testUtils/mockPersonResource.js [hunk 2]
```diff
@@ -29,14 +34,28 @@ export const aliceName = "Alice";
 export const aliceNick = "A";
 export const alicePhoto = "http://example.com/alice.jpg";
 
+const VCARD_WEBID_PREDICATE = "https://www.w3.org/2006/vcard/ns#WebId";
+
+export function mockWebIdNode(webId) {
+  const dataset = chain(
```
Comment (megoth) [inline __testUtils/mockPersonResource.js:40]: I would not call this `dataset`, since that can be confused with the type `SolidDataset`. Maybe `webIdNode` or `webIdThinng` would be better?

File: src/addressBook/index.js [hunk 3]
```diff
@@ -337,13 +369,13 @@ export function createContact(addressBookIri, contact, types) {
...
   );
-
-  const dataset = [...emails, ...addresses, ...telephones].reduce(
+  let dataset = [...emails, ...addresses, ...telephones].reduce(
     (acc, { thing }) => {
       return setThing(acc, thing);
     },
     person
   );
+  dataset = webIdNode && setThing(dataset, webIdNode);
```
Comment (megoth) [inline src/addressBook/index.js:378]: I'm a little concerned about this line, as it makes the overall logic a bit hard to follow for me. I reckon if `webIdNode` is falsy, it will simply return `false` for dataset? I cannot see any test that back up this scenario, or am I misunderstanding this?

**Review by megoth — CHANGES_REQUESTED** [review]
Just want to clarify a couple of things, then we should be good.

File: src/addressBook/index.js (hunk 3 above)
Comment (VirginiaBalseiro) [reply src/addressBook/index.js:378]: I can't remember exactly why I added that check tbh. Removed it since the webIdNode is never undefined (there's always a webId when we add a contact, and if it's invalid it will fail to fetch the profile and won't add the contact anyway).

**Review by kevin-inrupt — APPROVED** [review]

**Review by megoth — APPROVED** [review]

**Comment by megoth** [comment]
Btw, I forgot this earlier, but the ShEx shapes we must also be updated to reflect this change.

**Review by megoth — APPROVED** [review]
